internal/legacy/transform/transform.go

```diff
@@ -305,39 +306,85 @@ func ProcessEvidenceForFinding(vuln *definitions.Vulnerability, ev *testapi.Evid
 	return nil
 }
 
-// FindingToLegacyVuln is the beginning of the workflow in converting a snyk schema finding into
+// FindingToLegacyVulns is the beginning of the workflow in converting a snyk schema finding into
 // a legacy vulnerability to provide legacy json outputs.
-func FindingToLegacyVuln(finding *testapi.FindingData, logger *zerolog.Logger) (*definitions.Vulnerability, error) {
-	vuln := definitions.Vulnerability{Description: finding.Attributes.Description}
+func FindingToLegacyVulns(
+	finding *testapi.FindingData,
+	logger *zerolog.Logger,
+) ([]definitions.Vulnerability, error) {
+	baseVuln := definitions.Vulnerability{Description: finding.Attributes.Description}
 	for _, problem := range finding.Attributes.Problems {
-		err := ProcessProblemForVuln(&vuln, &problem, logger)
+		err := ProcessProblemForVuln(&baseVuln, &problem, logger)
 		if err != nil {
 			return nil, fmt.Errorf("handling problem for finding: %w", err)
 		}
 	}
 
 	for _, location := range finding.Attributes.Locations {
-		err := ProcessLocationForVuln(&vuln, &location, logger)
+		err := ProcessLocationForVuln(&baseVuln, &location, logger)
 		if err != nil {
 			return nil, fmt.Errorf("processing location for finding: %w", err)
 		}
 	}
 
-	vuln.From = []string{}
+	baseVuln.Title = finding.Attributes.Title
+	baseVuln.Severity = definitions.VulnerabilitySeverity(string(finding.Attributes.Rating.Severity))
+	if finding.Attributes.Risk.RiskScore != nil {
+		baseVuln.RiskScore = &finding.Attributes.Risk.RiskScore.Value
+	}
+
+	return processEvidences(finding, &baseVuln)
+}
 
+func processEvidences(
+	finding *testapi.FindingData,
+	baseVuln *definitions.Vulnerability,
+) ([]definitions.Vulnerability, error) {
+	var depPathEvidences []testapi.Evidence
+	var otherEvidences []testapi.Evidence
 	for _, ev := range finding.Attributes.Evidence {
-		err := ProcessEvidenceForFinding(&vuln, &ev)
+		evDisc, err := ev.Discriminator()
 		if err != nil {
-			return nil, fmt.Errorf("processing evidence for finding: %w", err)
+			return nil, fmt.Errorf("getting evidence discriminator: %w", err)
+		}
+		if evDisc == string(testapi.DependencyPath) {
+			depPathEvidences = append(depPathEvidences, ev)
+		} else {
+			otherEvidences = append(otherEvidences, ev)
 		}
 	}
-	vuln.Title = finding.Attributes.Title
-	vuln.Severity = definitions.VulnerabilitySeverity(string(finding.Attributes.Rating.Severity))
-	if finding.Attributes.Risk.RiskScore != nil {
-		vuln.RiskScore = &finding.Attributes.Risk.RiskScore.Value
+
+	var vulns []definitions.Vulnerability
+	if len(depPathEvidences) > 0 {
+		// Create a new legacy vulnerability for each dependency path evidence.
+		for _, depPathEv := range depPathEvidences {
+			vuln := *baseVuln
+			vuln.From = []string{}
```

Comment:
I feel like clearing the `From` belongs more in ProcessEvidenceForFinding than here.

It'd be preferrable IMO to build an entirely new slice there and replace any prior value of `From` with that new slice. The way ProcessEvidenceForFinding currently appends to the existing slice rather than building a new one unnecessarily dependent on an external state and thus more fragile.